### ISSUE TITLE
Report better error when `writeLine("stdout")` is used in a delay rule

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#7123]
- [ ] main
- [ ] 4-3-stable
---
## Bug Report
Using `"stdout"` as the target output device leads to a cryptic message with no helpful info.
```
{"log_category":"legacy","log_facility":"local0","log_level":"error","log_message":"error: syntax error\nline 0, col 1\n{\n ^\n\n","request_api_name":"EXEC_RULE_EXPRESSION_AN","request_api_number":1206,"request_api_version":"d","request_client_user":"rods","request_host":"XXXXX","request_proxy_user":"rods","request_release_version":"rods4.3.0","server_host":"XXX","server_pid":2099486,"server_timestamp":"2023-06-09T09:24:53.957Z","server_type":"agent"}
```

`writeLine()` or functions higher in the stack should detect this and report a good error message since the rule fails in an unexpected way.

_Originally reported at [https://groups.google.com/g/iROD-Chat/c/PhJ94yOE6t4](https://groups.google.com/g/iROD-Chat/c/PhJ94yOE6t4)._

**Comment by FifthPotato** [comment]
I'm having some troubles reproducing this on tip of main. I set up the rule exactly as the one in the original google group, and I can see it working-- made some separate rescs, repl'ed a file, set the metadata timestamp, and saw it get trimmed. However, complete silence from the log.

**Comment by trel** [comment]
Can you reproduce with 4.3.0?   Possible it's been 'fixed' along the way...

**Comment by FifthPotato** [comment]
> Can you reproduce with 4.3.0? Possible it's been 'fixed' along the way...

Redid everything with 4.3.0 and still can't see any weird log messages. Not sure if I'm missing something somewhere, but it's certainly hard to nail down.

**Comment by korydraughn** [comment]
And your tests included wrapping the call to `writeLine` in a delay rule?

**Comment by FifthPotato** [comment]
> And your tests included wrapping the call to `writeLine` in a delay rule?

Yep, exactly as written in the group. I can't think of anything else that could be causing this, so I'm going to say this can't be reproduced.

**Comment by korydraughn** [comment]
@FifthPotato Please close.
